### PR TITLE
Add pre-push hook example

### PR DESCRIPTION
**Pull request by nik-kor**

### CODE REVIEW

File: examples/pre-push/prevent-force-push-to-master [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+is_destructive='force|delete|\-f'
+
+will_remove_protected_branch=':'$protected_branch
+
+do_exit(){
+  echo $policy
+  exit 1
+}
+
+if [[ $push_command =~ $is_destructive ]] && [[ $push_command =~ $protected_branch ]]; then
```
Comment (tarmolov) [inline examples/pre-push/prevent-force-push-to-master]: It seems that two conditions can be merged into a single one and then `do_exit` function can be inlined.

File: examples/pre-push/prevent-force-push-to-master [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  do_exit
+fi
+
+if [[ $push_command =~ $will_remove_protected_branch ]]; then
+  do_exit
+fi
+
+unset do_exit
+
+exit 0
```
Comment (tarmolov) [inline examples/pre-push/prevent-force-push-to-master]: `exit 0` is a default behavior. It's better to omit it.

File: examples/pre-push/prevent-force-push-to-master [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+protected_branch='master'
+
+policy='[Policy] Never force push or delete the '$protected_branch' branch! (Prevented with pre-push hook.)'
+
+current_branch=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
+
+parent_pid=$(ps -oppid= -p $PPID)
+push_command=$(ps -ocommand= -p $parent_pid)
```
Comment (tarmolov) [inline examples/pre-push/prevent-force-push-to-master]: Can you explain what it is for? Looks a little bit weird)

File: examples/pre-push/prevent-force-push-to-master [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
```
Comment (tarmolov) [inline examples/pre-push/prevent-force-push-to-master]: Let's add a little description what this hook for.

File: examples/pre-push/prevent-force-push-to-master (hunk 3 above)
Comment (nik-kor) [inline examples/pre-push/prevent-force-push-to-master]: the current process is a node.js one. So to get the original git command we need to run ps with parent pid. It looks trivial or maybe I didn't understand the question

File: examples/pre-push/prevent-force-push-to-master (hunk 1 above)
Comment (nik-kor) [inline examples/pre-push/prevent-force-push-to-master]: done

File: examples/pre-push/prevent-force-push-to-master (hunk 2 above)
Comment (nik-kor) [inline examples/pre-push/prevent-force-push-to-master]: yep. done

File: examples/pre-push/prevent-force-push-to-master (hunk 4 above)
Comment (nik-kor) [inline examples/pre-push/prevent-force-push-to-master]: done

**Pull request merged**